WordPress/src/main/java/org/wordpress/android/ui/prefs/accountsettings/components/DialogErrorUi.kt

```diff
@@ -0,0 +1,87 @@
+package org.wordpress.android.ui.prefs.accountsettings.components
+
+import android.content.res.Configuration
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.material.ButtonDefaults
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import org.wordpress.android.R
+import org.wordpress.android.fluxc.network.rest.wpcom.account.CloseAccountResult.ErrorType
+import org.wordpress.android.fluxc.network.rest.wpcom.account.CloseAccountResult.ErrorType.UNAUTHORIZED
+import org.wordpress.android.fluxc.network.rest.wpcom.account.CloseAccountResult.ErrorType.ATOMIC_SITE
+import org.wordpress.android.fluxc.network.rest.wpcom.account.CloseAccountResult.ErrorType.CHARGEBACKED_SITE
+import org.wordpress.android.fluxc.network.rest.wpcom.account.CloseAccountResult.ErrorType.ACTIVE_SUBSCRIPTIONS
+import org.wordpress.android.fluxc.network.rest.wpcom.account.CloseAccountResult.ErrorType.ACTIVE_MEMBERSHIPS
+import org.wordpress.android.fluxc.network.rest.wpcom.account.CloseAccountResult.ErrorType.INVALID_TOKEN
+import org.wordpress.android.fluxc.network.rest.wpcom.account.CloseAccountResult.ErrorType.UNKNOWN
+import org.wordpress.android.ui.compose.theme.AppTheme
+
+@Composable
+fun DialogErrorUi(
+    onDismissRequest: () -> Unit,
+    onHelpRequested: () -> Unit,
+    errorType: ErrorType,
+) {
+    val padding = 10.dp
+    val messageId = when(errorType) {
+        UNAUTHORIZED -> R.string.account_closure_dialog_error_unauthorized
+        ATOMIC_SITE -> R.string.account_closure_dialog_error_atomic_site
+        CHARGEBACKED_SITE -> R.string.account_closure_dialog_error_chargebacked_site
+        ACTIVE_SUBSCRIPTIONS -> R.string.account_closure_dialog_error_active_subscriptions
+        ACTIVE_MEMBERSHIPS -> R.string.account_closure_dialog_error_active_memberships
+        INVALID_TOKEN, UNKNOWN -> R.string.account_closure_dialog_error_unknown
```

Comment:
This is a good question. The design goals here are to mirror the iOS implementation, which has these error messages, which I believe were obtained (and perhaps adapted?) from the web experience. In these circumstances, the user wants to close their account, but cannot do so without intervention from support folks (for reasons described in the messages themselves).

> Would there be cases where users don't end up contacting support because they figured out how to avoid the error by reading the message we display?

This is unlikely since they will still need to contact support, afaik.

One good reason for offering a less generic response in case of error is that the user in these cases will continue on to support to complete their desired action. The specific messages can provide better continuity in the experience, and help them reach their goals with less time and effort.